linetools/isgm/dla.py

```diff
@@ -0,0 +1,247 @@
+        zabs : float
+          Absorption redshift
+        vlim : Quantity array (2)
+          Velocity limits of the system
+          Defaulted to +/- 500 km/s if None
+        NHI : float, required despite being a keyword
+          log10 of HI column density
+          must be 20.3 or higher
+        **kwargs : keywords
+          passed to AbsSystem.__init__
+        """
+        # NHI
+        try:
+            NHI = kwargs['NHI']
+        except KeyError:
+            raise ValueError("NHI must be specified for LLSSystem")
+        else:
+            kwargs.pop('NHI')
+        if NHI < 20.3:
+            raise ValueError("This is not a DLA!  Try an LLS (or SLLS)")
+        # vlim
+        if vlim is None:
+            vlim = [-500., 500.]*u.km/u.s
+        # Generate with type
+        AbsSystem.__init__(self, 'DLA', radec, zabs, vlim, NHI=NHI, **kwargs)
+
+        # Other
+        self.ZH = 0.
+
+    def get_ions(self, use_clmfile=False, update_zvlim=True, linelist=None):
+        """Parse the ions for each Subsystem
+
+        And put them together for the full system
+        Fills ._ionN with a QTable
+
+        Parameters
+        ----------
+        idict : dict, optional
+          dict containing the IonClms info
+        use_clmfile : bool, optional
+          Parse ions from a .clm file (JXP historical)
+        update_zvlim : bool, optional
+          Update zvlim from lines in .clm (as applicable)
+        linelist : LineList
+        """
+        reload(ltiu)
+        if use_clmfile:
+            clm_fil = self.tree+self._datdict['Abund file']
+            # Read
+            self._clmdict = ltiu.read_clmfile(clm_fil, linelist=linelist)
+            # Build components
+            components = ltiu.build_components_from_abslines([], clmdict=self._clmdict, coord=self.coord)
+            # Read .ion file and fill in components
+            ion_fil = self.tree+self._clmdict['ion_fil']
+            self._indiv_ionclms = ltiu.read_ion_file(ion_fil, components)
+            # Parse .all file
+            all_file = ion_fil.split('.ion')[0]+'.all'
+            self.all_file=all_file #MF: useful to have
+            _ = ltiu.read_all_file(all_file, components=components)
+            # Build table
+            self._ionN = ltiu.iontable_from_components(components, ztbl=self.zabs)
+            # Add to AbsSystem
+            for comp in components:
+                self.add_component(comp)
+        else:
+            raise IOError("Not ready for this")
+
+    # Output
+    def __repr__(self):
+        return ('[{:s}: {:s} {:s}, {:g}, NHI={:g}, Z/H={:g}]'.format(
+                self.__class__.__name__,
+                 self.coord.ra.to_string(unit=u.hour, sep=':', pad=True),
+                 self.coord.dec.to_string(sep=':', pad=True),
+                 self.zabs, self.NHI, self.ZH))
+
+    def print_abs_type(self):
+        """"Return a string representing the type of vehicle this is."""
+        return 'DLA'
```

Comment:
Wouldn't this better as just an attribute called `abs_type`? Seems weird to make it a method.

(I know it's also in the base class GenericAbsorptionSystem, so it may be a bit late to be bringing this up...)